test/test_torch.py

```diff
@@ -2963,7 +2963,7 @@ def test_index_fill(self, device, dtype):
         index = torch.tensor([0], device=device)
         x.index_fill_(1, index, 0)
         self.assertEqual(x, torch.tensor([[0, 2], [0, 5]], dtype=dtype, device=device))
-        if not x.is_complex():
+        if not x.is_complex() and not device == "meta":
             with self.assertRaisesRegex(RuntimeError, r"Scalar"):
                 x.index_fill_(1, index, 1 + 1j)
```

Comment:
Hmm, actually, hypothetically, it should be possible for meta tensors to error here (since the meta tensor knows its not complex and thus writing in a complex should be rejected).

Reply:
I'll raise an issue. Do you want me to leave this here or skip / xfail the test for meta tensors until this is resolved?

Reply:
leave it here (i'm trying to land this anyhow), issue is great

Reply:
xref #73065